bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/threading/ThreadsafeSimpleRuleDelegate.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.automation.jsscripting.internal.threading;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.automation.Action;
+import org.openhab.core.automation.Condition;
+import org.openhab.core.automation.Module;
+import org.openhab.core.automation.Rule;
+import org.openhab.core.automation.Trigger;
+import org.openhab.core.automation.Visibility;
+import org.openhab.core.automation.module.script.rulesupport.shared.simple.SimpleRule;
+import org.openhab.core.automation.module.script.rulesupport.shared.simple.SimpleRuleActionHandler;
+import org.openhab.core.config.core.ConfigDescriptionParameter;
+import org.openhab.core.config.core.Configuration;
+
+/**
+ * An version of {@link SimpleRule} which controls multithreaded execution access to this specific rule. This is useful
+ * for rules which wrap GraalJS Contexts, which are not multithreaded.
+ *
+ * @author Jonathan Gilbert - Initial contribution
+ */
+@NonNullByDefault
+class ThreadsafeSimpleRuleDelegate implements Rule, SimpleRuleActionHandler {
+
+    private final Object lock;
+    private final SimpleRule delegate;
+
+    /**
+     * Constructor requires a lock object and delegate to forward invocations to.
+     *
+     * @param lock rule executions will synchronize on this object
+     * @param delegate the delegate to forward invocations to
+     */
+    ThreadsafeSimpleRuleDelegate(Object lock, SimpleRule delegate) {
+        this.lock = lock;
+        this.delegate = delegate;
+    }
+
+    @Override
+    @NonNullByDefault({})
+    public Object execute(Action module, Map<String, ?> inputs) {
+        synchronized (lock) {
+            return delegate.execute(module, inputs);
+        }
+    }
+
+    @Override
+    public String getUID() {
+        return delegate.getUID();
+    }
+
+    @Override
+    @Nullable
+    public String getTemplateUID() {
```

Comment:
```suggestion
    public @Nullable String getTemplateUID() {
```

Please also fix the others.